<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MqttClientWidget</class>
 <widget class="QWidget" name="MqttClientWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1024</width>
    <height>768</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Widget</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_5">
   <item row="0" column="0">
    <layout class="QHBoxLayout" name="horizontalLayout_17" stretch="1,4">
     <item>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QGroupBox" name="groupBox_3">
         <property name="title">
          <string>Setting</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_3">
          <item row="0" column="0">
           <layout class="QHBoxLayout" name="horizontalLayout" stretch="1,2">
            <item>
             <widget class="QLabel" name="label_serverIP">
              <property name="text">
               <string>Server IP</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="leHostaddr">
              <property name="text">
               <string/>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="1" column="0">
           <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="1,2">
            <item>
             <widget class="QLabel" name="label_port">
              <property name="text">
               <string>Port</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="lePort">
              <property name="text">
               <string/>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="2" column="0">
           <layout class="QHBoxLayout" name="horizontalLayout_3" stretch="1,2">
            <item>
             <widget class="QLabel" name="label_clientID">
              <property name="text">
               <string>Client ID</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="leClientId">
              <property name="text">
               <string/>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="3" column="0">
           <layout class="QHBoxLayout" name="horizontalLayout_9" stretch="1,2">
            <item>
             <widget class="QLabel" name="label_userName">
              <property name="text">
               <string>User Name</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="leUsernam">
              <property name="text">
               <string/>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="4" column="0">
           <layout class="QHBoxLayout" name="horizontalLayout_10" stretch="1,2">
            <item>
             <widget class="QLabel" name="label_password">
              <property name="text">
               <string>Password</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="lePassword">
              <property name="text">
               <string/>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="5" column="0">
           <layout class="QHBoxLayout" name="horizontalLayout_11" stretch="1,2">
            <item>
             <widget class="QLabel" name="label_keepAlive">
              <property name="text">
               <string>KeepAlive(s)</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="leKeepAlive">
              <property name="text">
               <string/>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="6" column="0">
           <layout class="QHBoxLayout" name="horizontalLayout_16" stretch="1,2">
            <item>
             <widget class="QLabel" name="label_status">
              <property name="text">
               <string>Status</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="pbConnectToHost">
              <property name="text">
               <string>Connect</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_sub">
         <property name="title">
          <string>Subscribe</string>
         </property>
         <layout class="QGridLayout" name="gridLayout">
          <item row="0" column="0">
           <layout class="QHBoxLayout" name="horizontalLayout_4">
            <item>
             <widget class="QLabel" name="label_subscribeTopic">
              <property name="text">
               <string>Topic</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="leSubscribeTopic"/>
            </item>
           </layout>
          </item>
          <item row="1" column="0">
           <layout class="QHBoxLayout" name="horizontalLayout_12" stretch="1,1">
            <item>
             <widget class="QLabel" name="label_subQos">
              <property name="text">
               <string>Qos</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="cbxQosSubscribe"/>
            </item>
           </layout>
          </item>
          <item row="2" column="0">
           <layout class="QHBoxLayout" name="horizontalLayout_13" stretch="1,1">
            <item>
             <widget class="QPushButton" name="pbClearRecv">
              <property name="text">
               <string>Clear Data</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="pbSubscribeTopic">
              <property name="text">
               <string>Subscribe</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="3" column="0">
           <layout class="QHBoxLayout" name="horizontalLayout_5" stretch="1,1">
            <item>
             <widget class="QComboBox" name="cbxSubscribedTopic"/>
            </item>
            <item>
             <widget class="QPushButton" name="pbUnsubscribeTopic">
              <property name="text">
               <string>UnSubscribe</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_pub">
         <property name="title">
          <string>Publish</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_2">
          <item row="0" column="0">
           <layout class="QHBoxLayout" name="horizontalLayout_6">
            <item>
             <widget class="QLabel" name="label_publishTopic">
              <property name="text">
               <string>Topic</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="lePublishTopic"/>
            </item>
           </layout>
          </item>
          <item row="1" column="0">
           <layout class="QHBoxLayout" name="horizontalLayout_14" stretch="1,1">
            <item>
             <widget class="QLabel" name="label_pubQos">
              <property name="text">
               <string>Qos</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="cbxQosPublish"/>
            </item>
           </layout>
          </item>
          <item row="2" column="0">
           <layout class="QHBoxLayout" name="horizontalLayout_7">
            <item>
             <widget class="QLabel" name="label_pubData">
              <property name="text">
               <string>Data</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QTextEdit" name="tePublicData"/>
            </item>
           </layout>
          </item>
          <item row="3" column="0">
           <layout class="QHBoxLayout" name="horizontalLayout_8" stretch="1,1">
            <item>
             <widget class="QPushButton" name="pbClearPublish">
              <property name="text">
               <string>Clear Data</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="pbPublishTopic">
              <property name="text">
               <string>Publish</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="4" column="0">
           <widget class="QCheckBox" name="checkBox_autoSend">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>AutoSend</string>
            </property>
           </widget>
          </item>
          <item row="5" column="0">
           <layout class="QHBoxLayout" name="horizontalLayout_15">
            <item>
             <widget class="QLabel" name="label_autoSend">
              <property name="text">
               <string>Interval</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="spinBox_autoSendInterval">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_ms">
              <property name="text">
               <string>ms</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_2" stretch="1,0">
       <item>
        <widget class="QGroupBox" name="groupBox_4">
         <property name="title">
          <string>Data Rx</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_4">
          <item row="0" column="0">
           <widget class="QTextEdit" name="teRecvData"/>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_18">
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_3" stretch="1,1,1">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_19" stretch="1,1">
             <item>
              <widget class="QLabel" name="label_rxPacket">
               <property name="text">
                <string>Rx Packet:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="label_rxPacketCnt">
               <property name="text">
                <string>0</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_20" stretch="1,1">
             <item>
              <widget class="QLabel" name="label_rxPacketAvg">
               <property name="text">
                <string>Rx Packet Average:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="label_rxPacketCntAvg">
               <property name="text">
                <string>0</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_21" stretch="1,1">
             <item>
              <widget class="QLabel" name="label_rxBytes">
               <property name="text">
                <string>Rx Bytes:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="label_rxBytesCnt">
               <property name="text">
                <string>0</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_5" stretch="1,1,1">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_22" stretch="1,1">
             <item>
              <widget class="QLabel" name="label_txPacket_2">
               <property name="text">
                <string>Tx Packet:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="label_txPacketCnt">
               <property name="text">
                <string>0</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_23" stretch="1,1">
             <item>
              <widget class="QLabel" name="label_txPacket_3">
               <property name="text">
                <string>Tx Packet Average:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="label_txPacketCntAvg">
               <property name="text">
                <string>0</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_24" stretch="1,1">
             <item>
              <widget class="QLabel" name="label_txBytes">
               <property name="text">
                <string>Tx Bytes:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="label_txBytesCnt">
               <property name="text">
                <string>0</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </item>
         <item>
          <spacer name="horizontalSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QPushButton" name="pushButton_reset">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Reset</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
